# HG changeset patch
# User Rik <rik@octave.org>
# Date 1551835863 28800
#      Tue Mar 05 17:31:03 2019 -0800
# Node ID 98f34c841f89a7da774cfa1dffb2cd09d556e8aa
# Parent  603f5d6ada56a5c30cb96ce160513b37a46ec52b
# Parent  4d7665fc8f6c797ad085ade5f346da346e1a0778
maint: merge stable to default.

diff --git a/doc/interpreter/package.txi b/doc/interpreter/package.txi
--- a/doc/interpreter/package.txi
+++ b/doc/interpreter/package.txi
@@ -353,18 +353,22 @@ License: GPLv3+
 
 The package manager currently recognizes the following keywords
 
 @table @code
 @item Name
 Name of the package.
 
 @item Version
-Version of the package.  A package version must be 3 numbers separated
-by dots.
+Version of the package.  A package version is typically digits separated by
+dots but may also contain @samp{+}, @samp{-}, @samp{~}, and alphanumeric
+characters (in the "C" locale).  For example, @qcode{"2.1.0+"} could indicate
+a development version of a package.
+@c regexp in get_description.m:is_valid_pkg_version_string
+Versions are compared using @ref{XREFcompare_versions,,compare_versions}.
 
 @item Date
 Date of last update.
 
 @item Author
 Original author of the package.
 
 @item Maintainer
diff --git a/etc/module.mk b/etc/module.mk
--- a/etc/module.mk
+++ b/etc/module.mk
@@ -49,16 +49,17 @@ desktopdir = $(datadir)/applications
 desktop_DATA = \
   %reldir%/icons/org.octave.Octave.desktop
 
 icon_IMAGE_FILES = \
   %reldir%/icons/octave-logo.svg \
   %reldir%/icons/octave-sombrero.png
 
 icon_PNG_SIZES = \
+  1024 \
   512 \
   256 \
   128 \
   64 \
   48 \
   32 \
   24 \
   22 \
diff --git a/scripts/plot/draw/line.m b/scripts/plot/draw/line.m
--- a/scripts/plot/draw/line.m
+++ b/scripts/plot/draw/line.m
@@ -63,21 +63,30 @@
 
 function h = line (varargin)
 
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("line", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
+    oldfig = [];
   else
     hax = hax(1);
+    oldfig = get (0, "currentfigure");
+    set (0, "currentfigure", ancestor (hax, "figure"));
   endif
 
-  htmp = __line__ (hax, varargin{:});
+  unwind_protect
+    htmp = __line__ (hax, varargin{:});
+  unwind_protect_cleanup
+    if (! isempty (oldfig))
+      set (0, "currentfigure", oldfig);
+    endif
+  end_unwind_protect
 
   if (nargout > 0)
     h = htmp;
   endif
 
 endfunction
 
 
